scargo/scargo_src/sc_src.py

```diff
@@ -482,41 +373,21 @@ def check_clang_tidy(config: Config, verbose: bool) -> None:
         exclude_list.extend(glob.glob(pattern))
 
     # Collect local excludes.
-    for pattern in config.check.clang_tidy.exclude:
+    for pattern in exclude_patterns:
         exclude_list.extend(glob.glob(pattern))
 
-    for root, _, filenames in os.walk(source_dir):
-        for fname in filenames:
-            fname = os.path.join(root, fname)
-
-            exist_flag = False
-            for exclude in exclude_list:
-                if exclude in fname:
-                    logger.info("Skipping %s", fname)
-                    exist_flag = True
-                    break
-            if exist_flag:
+    glob_patterns = (
+        ("*.h", "*.hpp") if headers_only else ("*.h", "*.hpp", "*.c", "*.cpp")
+    )
+    for file_path in chain.from_iterable(
```

Comment:
Doesn't this raise complexity a lot or am I missing something? Maybe os.walk is better option for this kind of use case

Reply:
Do you mean time or space complexity or some other kind of complexity? https://en.wikipedia.org/wiki/Computational_complexity

Reply:
As for time complexity, it does not.

Reply:
Time, I remember doing something like this and benchmarking it eventually and os.walk ended up being a lot faster in my case.

Reply:
Found the example where I benchmarked this. I didn't go into details why this is happening, but I suspect it's creation of Path objects for each file. It's not 1on1 match with your code, but you can take a look:
![image](https://user-images.githubusercontent.com/123727166/221176915-648822cd-5c22-4bb2-b797-2cde4c224cc6.png)

Reply:
Ok, maybe it's a bit slower, but don't confuse it with time complexity. It's still the same (linear).

Reply:
I've checked it on my copy of the scargo repository and `os.walk` was only ~2.4 times slower (and we're talking about really negligible times, like less than 0.1 s, so it's no real concern.
On the other hand, with `rglob` I can avoid calling `os.path.join` which was the original goal.